soap/src/test/java/feign/soap/SOAPFaultDecoderTest.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright 2012-2018 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign.soap;
+
+import static feign.Util.UTF_8;
+import java.io.IOException;
+import java.util.Collections;
+import javax.xml.soap.SOAPConstants;
+import javax.xml.ws.soap.SOAPFaultException;
+import org.assertj.core.api.Assertions;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import feign.FeignException;
+import feign.Request;
+import feign.Request.HttpMethod;
+import feign.jaxb.JAXBContextFactory;
+import feign.soap.SOAPDecoder;
+import feign.soap.SOAPErrorDecoder;
+import feign.Response;
+import feign.Util;
+
+public class SOAPFaultDecoderTest {
+
+  private static final String SOAP_1_2_FAULT = "<?xml version=\"1.0\"?>\n" +
```

Comment:
It would be nice (for maintenance) to move this to files inside `src/test/resources/samples`

Reply:
Hi Velo,
I externalized the SOAP Envelop in corresponding XML files.